server/pulp/plugins/rsync/publish.py

```diff
@@ -378,30 +364,11 @@ def is_fastforward(self):
         :rtype: bool
         """
         force_full = False
-        for entry in self.predist_history:
-            predistributor_force_full = entry.get("distributor_config", {}).get("force_full",
-                                                                                False)
-            force_full |= predistributor_force_full
-            if entry.get("result", "error") == "error":
-                force_full = True
-
-        if self.last_published:
-            last_published = self.last_published.replace(tzinfo=None)
-        else:
-            last_published = None
-
-        if self.last_deleted:
```

Comment:
it feels like a sane logic and rsync should not happen in a force full mode if a unit got removed however I am afraid we might miss some other corner case. @rohanpm can you please review this PR? I feel like you would have the best insight here since my brain needed to be cleaned  from a layer of dust to remember this area.

Reply:
Yeah, I think this is right.  We looked into the history of this code and discussed it internally prior to this PR.